### PR TITLE
CVSL-1429 Fixing release event handler HDC logic

### PR DESCRIPTION
**Pull request by andrewrlee**
Only inactivate for HDC if there is a valid HDCED

### CODE REVIEW

File: server/data/licenceApiClient.test.ts [hunk 1]
```diff
@@ -270,11 +270,6 @@ describe('Licence API client tests', () => {
     )
   })
 
-  it('Batch activate licences', async () => {
-    await licenceApiClient.batchActivateLicences([123, 321])
-    expect(post).toHaveBeenCalledWith({ path: '/licence/activate-licences', data: [123, 321] })
-  })
```
Comment (andrewrlee) [inline server/data/licenceApiClient.test.ts:276]: not used

File: server/listeners/eventHandlers/domainEvents/releaseEventHandler.test.ts [hunk 2]
```diff
@@ -151,10 +147,7 @@ describe('Release event handler', () => {
         restrictedPatient: false,
       },
     ])
-    prisonerService.getActiveHdcStatus.mockResolvedValue({
-      approvalStatus: 'APPROVED',
-      bookingId: '111',
-    })
+    prisonerService.isHdcApproved.mockResolvedValue(true)
```
Comment (andrewrlee) [inline server/listeners/eventHandlers/domainEvents/releaseEventHandler.test.ts:150]: this was a bit horrible - it was coupled to the internals of the prisoner service as this method wasn't mocked. 

I've moved the prisonerservice over to be a simple mock and this now solely tests the interactions.
I've added proper unit tests to filling in missing coverage for this `isHdcApproved` method

File: server/services/prisonerService.ts [hunk 3]
```diff
@@ -127,8 +125,12 @@ export default class PrisonerService {
     )
   }
 
-  async hdcLicenceIsApproved(bookingId: string): Promise<boolean> {
-    const hdcLicence = await this.getActiveHdcStatus(bookingId)
+  async isHdcApproved(bookingId: number): Promise<boolean> {
+    const prisoners = await this.prisonerSearchApiClient.searchPrisonersByBookingIds({ bookingIds: [bookingId] })
+    if (!prisoners.length || !prisoners[0].homeDetentionCurfewActualDate) {
+      return false
```
Comment (andrewrlee) [inline server/services/prisonerService.ts:131]: they can no longer be classed for HDC if they are no longer eligible, regardless what NOMIS says.

**Review by justin-gardiner — DISMISSED** [review]
lgtm

**Review by justin-gardiner — COMMENTED** [review]
got a failing test

**Review by justin-gardiner — APPROVED** [review]
lgtm

**Pull request merged**